examples_tests/48.ArithmeticUnitTest/main.cpp

```diff
@@ -0,0 +1,230 @@
+		{TO_AND,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_XOR,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_OR,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_ADD,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_MUL,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_MIN,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+		{TO_MAX,EDT_STORAGE_BUFFER,1u,IGPUSpecializedShader::ESS_COMPUTE,nullptr},
+	};
+	auto gpuDSLayout = driver->createGPUDescriptorSetLayout(binding, binding + 8);
+	constexpr uint32_t pushconstantSize = 1;
+	SPushConstantRange pcRange[1] = { IGPUSpecializedShader::ESS_COMPUTE,0u,pushconstantSize };
+	auto pipelineLayout = driver->createGPUPipelineLayout(pcRange, pcRange + pushconstantSize, core::smart_refctd_ptr(gpuDSLayout));
+
+	auto descriptorSet = driver->createGPUDescriptorSet(core::smart_refctd_ptr(gpuDSLayout));
+	{
+		IGPUDescriptorSet::SDescriptorInfo infos[8];
+		infos[0].desc = gpuinputDataBuffer;
+		infos[0].buffer = { 0u, BUFFER_SIZE };
+		for (size_t i = 1; i <= 7; i++)
+		{
+			infos[i].desc = buffers[i - 1];
+			infos[i].buffer = { 0u,BUFFER_SIZE };
+
+		}
+		IGPUDescriptorSet::SWriteDescriptorSet writes[8];
+		for (uint32_t i = 0u; i < 8; i++)
+			writes[i] = { descriptorSet.get(),i,0u,1u,EDT_STORAGE_BUFFER,infos + i };
+		driver->updateDescriptorSets(8, writes, 0u, nullptr);
+	}
+	struct GLSLCodeWithWorkgroup {
+		uint32_t workgroup_definition_position;
+		std::string glsl;
+	};
+	constexpr const char* symbolsToReplace = "????";
+	auto getShaderGLSL = [&](const char* filePath)
+	{
+		std::ifstream file(filePath);
+		std::stringstream buff; buff << file.rdbuf();
+		std::string shaderCode = buff.str();
+		uint32_t wgPos = shaderCode.find(symbolsToReplace, 0);
+		GLSLCodeWithWorkgroup ret = { wgPos,shaderCode };
+		return ret;
+	};
+	GLSLCodeWithWorkgroup shaderGLSL[3] =
+	{
+		getShaderGLSL("../testReduce.comp"),
+		getShaderGLSL("../testInclusive.comp"),
+		getShaderGLSL("../testExclusive.comp"),
+	};
+
+
+auto getGPUShader = [&](GLSLCodeWithWorkgroup glsl, uint32_t wg_count)
+{
+	auto alteredGLSL = glsl.glsl.replace(glsl.workgroup_definition_position, 4, std::to_string(wg_count));
+	auto shaderUnspecialized = core::make_smart_refctd_ptr<asset::ICPUShader>(alteredGLSL.data());
+	asset::ISpecializedShader::SInfo specinfo(nullptr, nullptr, "main", IGPUSpecializedShader::ESS_COMPUTE,"../file.comp");
+	auto cs = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(shaderUnspecialized), std::move(specinfo));
+	auto cs_rawptr = cs.get();
+	core::smart_refctd_ptr<IGPUSpecializedShader> shader = driver->getGPUObjectsFromAssets(&cs_rawptr, &cs_rawptr + 1)->front();
+	return shader;
+};
+	for (size_t current_Workgroup = 2; current_Workgroup < 1024; current_Workgroup++)
```

Comment:
leave a comment that subgroup size is hardcoded to 4 (there's a requirement that workgroup be at least half the size) for now and max workgroup size is hardcoded to 256